### PR TITLE
Update DiscordRPC to plugins v4

### PR DESCRIPTION
**Pull request by catornot**
:)
100% rust now

maybe I will actually move everything into the plugin soon

### CODE REVIEW

**Review by GeckoEidechse — APPROVED** [review]
Confirmed working in testing together with https://github.com/R2Northstar/NorthstarLauncher/pull/615

Launched the game and joined a PvP and a Parkour server each while checking that Discord activity would update accordingly.

File: src/discord.rs [hunk 1]
```diff
@@ -14,7 +14,8 @@ use rrplug::prelude::*;
 use crate::exports::PLUGIN;
 
 /// the discord app's id, taken from older v1 discord rpc
-const APP_ID: i64 = 941428101429231617;
+// const APP_ID: i64 = 941428101429231617;
+const APP_ID: i64 = 1198009103516180500;
```
Comment (uniboi) [inline src/discord.rs]: did the ID change?

Comment (catornot) [reply src/discord.rs]: Yeah, I want mp_box to be included

Comment (GeckoEidechse) [reply src/discord.rs]: Should be a separate PR IMO ^^"

Comment (catornot) [reply src/discord.rs]: removed it

**Review by uniboi — APPROVED** [review]
looks good, really just an update to a newer rrplug commit

File: manifest/manifest.json [hunk 2]
```diff
@@ -1,10 +0,0 @@
-{
-	"name": "Northstar.DiscordRPC",
-	"displayname": "DPRESENCE",
-	"dependencyName": "DISCORDRPC",
-	"description": "Discord Rich Presence for Northstar",
-	"api_version": "3",
-	"version": "3.0",
```
Comment (GeckoEidechse) [inline manifest/manifest.json:7]: With `manifest.json` being gone, we should probably set version number in `Cargo.toml` correctly to v4 ^^

https://github.com/R2Northstar/NorthstarDiscordRPC/pull/17/files#diff-2e9d962a08321605940b5a657135052fbcef87b5e360662bb527c96d9a615542R3

**Review by GeckoEidechse — APPROVED** [review]
Retested as working with https://github.com/R2Northstar/NorthstarLauncher/pull/615

c.f.: https://github.com/R2Northstar/NorthstarLauncher/pull/615#pullrequestreview-1861170788

File: manifest/manifest.json (hunk 2 above)
Comment (catornot) [reply manifest/manifest.json:7]: done

File: src/lib.rs [hunk 3]
```diff
@@ -38,8 +40,15 @@ pub struct DiscordRpcPlugin {
 
 #[deny(non_snake_case)]
 impl Plugin for DiscordRpcPlugin {
-    fn new(plugin_data: &PluginData) -> Self {
-        plugin_data.register_sq_functions(presence::fetch_presence);
+    const PLUGIN_INFO: PluginInfo = PluginInfo::new(
+        "DISCORDRPC\0",
+        "DISCORDXD\0",
```
Comment (GeckoEidechse) [inline src/lib.rs]: I know `log_name` being shorter is so it matches the rest in length but why the `XD` suffix. Just cause `xD` or am I missing some obvious abbreviation here? ^^


If not, you could also do something like `DSCRD-RPC` ig

Comment (catornot) [reply src/lib.rs]: I just couldn't a come up with anything better better than `XD` suffix.
`DSCRD-RPC` is definitely better.

**Review by GeckoEidechse — COMMENTED** [review]
Everything from reviews has been addressed. This should be good to merge now ^^

**Pull request merged**